test/unittest/unittest_nntrainer_tensor_fp16.cpp

```diff
@@ -6251,6 +6251,34 @@ TEST(nntrainer_Tensor, TensorWrap_02_n) {
 //   EXPECT_EQ(output, answer3);
 // }
 
+/**
+ * @brief fp16 tensor has NaN
+ */
+TEST(nntrainer_Tensor, is_valid_01) {
+  size_t batch = 1;
+  size_t channel = 3;
+  size_t height = 4;
+  size_t width = 5;
+
+  nntrainer::Tensor input(
+    {batch,
+     channel,
+     height,
+     width,
+     {nntrainer::Tformat::NCHW, nntrainer::Tdatatype::FP16}},
+    true, nntrainer::Initializer::ZEROS);
+
+  EXPECT_EQ(input.isValid(), true);
+
+  input.setValue(0, 0, 0, 0, std::nan("1"));
+
+  EXPECT_EQ(input.isValid(), false);
+
+  input.setValue(0, 0, 0, 0, std::numeric_limits<float>::infinity());
+
+  EXPECT_EQ(input.isValid(), false);
+}
+
```

Comment:
Is there any difference w.r.t. `TEST(nntrainer_Tensor, is_valid_01) {...}` at `test/unittest/unittest_nntrainer_tensor_neon_fp16.cpp`?
I think this is a duplicated TC.
SIMD and non-SIMD TC can be differentiated by flags, not by TC I guess?

Reply:
Hmm...  I think FP16 can be enabled without neon options. If this can be tested with and without 'enable_neon' then we could remove the test case in neon_fp16.cpp.